### DISCUSSION TITLE
Does USFM have strict rules about spacing and punctuations?

### DISCUSSION THREAD

**Discussion by kavitharaju** [usfm-bible/tcdocs#30] [General]
Does these paratext specific rules, in the test suite, apply as such to the USFM spec?
- [paratextTests/WordlistMarkerTextEndsInSpaceWithoutGlossary](https://github.com/usfm-bible/tcdocs/blob/471a15103622813ddd4e4eb13b17bb464bf02cbf/tests/paratextTests/WordlistMarkerTextEndsInSpaceWithoutGlossary/metadata.xml#L4) should fail
- [paratextTests/WordlistMarkerTextEndsInSpaceWithGlossary](https://github.com/usfm-bible/tcdocs/blob/471a15103622813ddd4e4eb13b17bb464bf02cbf/tests/paratextTests/WordlistMarkerTextEndsInSpaceWithGlossary/metadata.xml#L4)  should fail
- [paratextTests/WordlistMarkerTextContainsNonWordformingPunctuation](https://github.com/usfm-bible/tcdocs/blob/471a15103622813ddd4e4eb13b17bb464bf02cbf/tests/paratextTests/WordlistMarkerTextContainsNonWordformingPunctuation/metadata.xml#L4) should fail
- [paratextTests/GlossaryCitationFormContainsNonWordformingPunctuation](https://github.com/usfm-bible/tcdocs/blob/471a15103622813ddd4e4eb13b17bb464bf02cbf/tests/paratextTests/GlossaryCitationFormContainsNonWordformingPunctuation/metadata.xml#L4) should fail
- [paratextTests/WordlistMarkerTextEndsInPunctuation](https://github.com/usfm-bible/tcdocs/blob/471a15103622813ddd4e4eb13b17bb464bf02cbf/tests/paratextTests/WordlistMarkerTextEndsInPunctuation/metadata.xml#L4)  should fail
- [paratextTests/GlossaryCitationFormEndsInSpace](https://github.com/usfm-bible/tcdocs/blob/471a15103622813ddd4e4eb13b17bb464bf02cbf/tests/paratextTests/GlossaryCitationFormEndsInSpace/metadata.xml#L4)  should fail
- [paratextTests/WordlistMarkerKeywordEndsInSpace](https://github.com/usfm-bible/tcdocs/blob/471a15103622813ddd4e4eb13b17bb464bf02cbf/tests/paratextTests/WordlistMarkerKeywordEndsInSpace/metadata.xml)  should fail
- [paratextTests/WordlistMarkerKeywordEndsInPunctuation](https://github.com/usfm-bible/tcdocs/blob/471a15103622813ddd4e4eb13b17bb464bf02cbf/tests/paratextTests/WordlistMarkerKeywordEndsInPunctuation/metadata.xml)  should fail
- [paratextTests/GlossaryCitationFormEndsInPunctuation](https://github.com/usfm-bible/tcdocs/blob/471a15103622813ddd4e4eb13b17bb464bf02cbf/tests/paratextTests/GlossaryCitationFormEndsInPunctuation/metadata.xml#L4)  should fail
- [paratextTests/WordlistMarkerTextEndsInSpaceAndMissingFromGlossary](https://github.com/usfm-bible/tcdocs/blob/471a15103622813ddd4e4eb13b17bb464bf02cbf/tests/paratextTests/WordlistMarkerTextEndsInSpaceAndMissingFromGlossary/metadata.xml#L4)  should fail
- [paratextTests/WordlistMarkerKeywordContainsNonWordformingPunctuation](https://github.com/usfm-bible/tcdocs/blob/471a15103622813ddd4e4eb13b17bb464bf02cbf/tests/paratextTests/WordlistMarkerKeywordContainsNonWordformingPunctuation/metadata.xml#L4)  should fail
 
Also, can it be said closing and opening a character marker immediately an incorrect usage?
- https://github.com/usfm-bible/tcdocs/blob/471a15103622813ddd4e4eb13b17bb464bf02cbf/tests/paratextTests/CharStyleClosedAndReopened/origin.usfm#L3

**Comment by KentSpiel** [comment]
USFM specs do not specify anything about text content. There are no strict rules about spacing and punctuation. Closing and opening a character marker immediately is allowed.
